### PR TITLE
Implement retrieving existing token

### PR DESCRIPTION
**Pull request by kimlisa**
task: https://github.com/hwsc-org/hwsc-user-svc/issues/58

- updated GetToken to retrieve existing token if user is already present in token table
- implemented and unit tested getExistingToken
- moved unit test utility functions to own file called consts_test.go
- linted and formatted
- ran all unit tests

screen shot 2019-02-18 at 9 04 52 pm
screen shot 2019-02-18 at 9 04 59 pm

### CODE REVIEW

File: service/db.go [hunk 1]
```diff
@@ -634,3 +636,53 @@ func insertJWToken(token string, header *auth.Header, body *auth.Body, secret *p
...
+			token:      token,
+			secret: &pblib.Secret{
+				Key:                 secret,
+				CreatedTimestamp:    secretCreatedTimestamp.Unix(),
+				ExpirationTimestamp: secretExpirationTimestamp.Unix(),
+			},
+		}, nil
+	}
+
+	return nil, nil
```
Comment (faraonc) [inline service/db.go]: This needs to return an error. I understand that returning `nil` can be ideal, but I would rather return an error here.

`if err != nil` then make a new token

File: service/service.go [hunk 2]
```diff
@@ -513,6 +513,22 @@ func (s *Service) GetToken(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc.
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	// lookup user in token table for existing user
+	existingToken, err := getExistingToken(retrievedUser.GetUuid())
+	if err != nil {
+		logger.Error(consts.GetAuthTokenTag, consts.MsgErrGetExistingToken, err.Error())
+	}
+	if existingToken != nil && existingToken.permission == retrievedUser.GetPermissionLevel() {
```
Comment (faraonc) [inline service/service.go]: If you made `getExistingToken` return an `error` on the last `return statement`, then you would not need the condition `existingToken != nil`

Comment (kimlisa) [reply service/service.go]: i pushed changes, not sure if the change is what you meant

File: service/service.go [hunk 3]
```diff
@@ -513,39 +513,51 @@ func (s *Service) GetToken(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc.
...
-	}
-	body := &auth.Body{
-		UUID:                retrievedUser.GetUuid(),
-		Permission:          permissionLevel,
-		ExpirationTimestamp: time.Now().UTC().Add(time.Hour * time.Duration(jwtExpirationTime)).Unix(),
-	}
+	existingToken, err := getExistingToken(retrievedUser.GetUuid())
+	if err != nil || (existingToken != nil && existingToken.permission != retrievedUser.GetPermissionLevel()) {
+		// TODO include mapping table in db
+		permissionLevel := auth.PermissionEnumMap[retrievedUser.GetPermissionLevel()]
```
Comment (faraonc) [inline service/service.go]: Your algorithm should be:
`if existingToken.permission != retrievedUser.GetPermissionLevel()`, then do not proceed further as discussed from the comment above.

`if err != nil`, then log  the error, and make a new token.

**Pull request merged**